cnctcli/actions/products/media_sync.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of the Ingram Micro Cloud Blue Connect connect-cli.
+# Copyright (c) 2019-2020 Ingram Micro. All Rights Reserved.
+
+from collections import namedtuple
+
+from tqdm import trange
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+from cnctcli.actions.products.sync import ProductSynchronizer
+from cnctcli.actions.products.constants import MEDIA_COLS_HEADERS
+from cnct import ClientError
+
+from urllib.parse import urlparse
+
+import os
+
+
+fields = (v.replace(' ', '_').lower() for v in MEDIA_COLS_HEADERS.values())
+
+_RowData = namedtuple('RowData', fields)
+
+
+class MediaSynchronizer(ProductSynchronizer):
+    def __init__(self, client, silent):
+        self._media_path = None
+        super(MediaSynchronizer, self).__init__(client, silent)
+
+    def open(self, input_file, worksheet):
+        self._media_path = input_file.rsplit('/', 1)[0]
+        super(MediaSynchronizer, self).open(input_file, worksheet)
+
+    def sync(self):
+        ws = self._wb['Media']
+        errors = {}
+        skipped_count = 0
+        created_items = []
+        updated_items = []
+        deleted_items = []
+
+        row_indexes = trange(2, ws.max_row + 1, position=0, disable=self._silent)
+        for row_idx in row_indexes:
+            data = _RowData(*[ws.cell(row_idx, col_idx).value for col_idx in range(1, 7)])
+            row_indexes.set_description(f'Processing Media {data.id or data.position or "New"}')
+
+            if data.action == '-':
+                skipped_count += 1
+                continue
+            row_errors = self._validate_row(data)
+
+            if row_errors:
+                errors[row_idx] = row_errors
+                continue
+            if data.action == 'delete':
+                try:
+                    self._client.products[self._product_id].media[data.id].delete()
+                except ClientError:
```

Comment:
Same here, If the ClientError is raised due to a timeout for example, the media has not been deleted but is counted as it was